### PR TITLE
Necroa's salvage update

### PR DESCRIPTION
**Pull request by Necro-a**
## About the PR
 Added a new Salvage facility: Geneticore, for the Shadow biome, as well as 2 new enemy types, Flesh and Elementals. Also added more drops to the expedition table. I also switched the Shadow biomes chasms for shadow anomaly shadows.

## Why / Balance
 I have always felt like salvage expeditions needed more variety and more exciting loot.

## Technical details

Negative, sir.
## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
Skärmbild 2024-05-17 162656
Skärmbild 2024-05-17 162438

## Breaking changes

**Changelog**

Added Geneticore facility
Reworked the shadow biome by replacing the chasms with shadow anomaly shadows
Added more expedition drops
Added Ore crabs and Flesh monsters to expedition enemy types.

### CODE REVIEW

**Comment by Simyon264** [comment]
Could you please post screenshots of the additions?

**Comment by Necro-a** [comment]
> Could you please post screenshots of the additions?

Is that good?

**Comment by Necro-a** [comment]
Also, small note, this is the first time im doing something like this, keep that in mind

**Comment by TsjipTsjip** [comment]
Something about this pull request is upsetting the linter. I'll have to poke around with the PR to see what exactly is going on.

**Comment by Kukutis96513** [comment]
how rare is this facility anyway? looking at the screenshots theres alot of loot in it.

**Comment by Necro-a** [comment]
> Something about this pull request is upsetting the linter. I'll have to poke around with the PR to see what exactly is going on.

Fixed it, added the tag needed.

File: Resources/Maps/Dungeon/geneticore.yml
Comment (TsjipTsjip) [inline Resources/Maps/Dungeon/geneticore.yml:1]: Should be in the _Umbra namespace.

File: Resources/Prototypes/Entities/Mobs/NPCs/flesh.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Entities/Mobs/NPCs/flesh.yml:1]: Should be added in a new file in the _Umbra namespace.

File: Resources/Prototypes/Procedural/Themes/geneticore.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/Themes/geneticore.yml:1]: Should be in the _Umbra namespace.

File: Resources/Prototypes/Procedural/biome_templates.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/biome_templates.yml:1]: What do these changes do in-game? This looks like it affects things outside of the salvage dungeon being added.

File: Resources/Prototypes/Procedural/dungeon_configs.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/dungeon_configs.yml:1]: Should be added in a new file in the _Umbra namespace.

File: Resources/Prototypes/Procedural/salvage_loot.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/salvage_loot.yml:1]: Umbra comment implies only WeaponCrusher was added. Put Umbra comments on everything, or make it clear by stating `# START OF UMBRA ADDITIONS` and `# END OF UMBRA ADDITIONS` around the entire modified block.

This is gonna really really need a big balancing consideration as this also impacts other dungeons. I'm not sure I like this at first glance but we'll see how we go along.

File: Resources/Prototypes/Procedural/salvage_mods.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/salvage_mods.yml:1]: Separate file in _Umbra namespace, or Umbra comments.

File: Resources/Prototypes/tags.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/tags.yml:1]: No clue what this does, not my expertise, needs an _Umbra namespaced file tho.

File: Resources/Prototypes/Procedural/salvage_factions.yml
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/salvage_factions.yml:1]: _Umbra namespace please.

**Review by TsjipTsjip — CHANGES_REQUESTED** [review]
Lots of the same stuff. Have not tested this in-game yet. (How-to-test steps would be appreciated.)

File: Resources/Prototypes/Procedural/biome_templates.yml
Comment (Necro-a) [reply Resources/Prototypes/Procedural/biome_templates.yml:1]: In game, all this does is replace the shadow biomes chasms with shadow kudzu shadows. i found it alot more fitting and less annoying to play. Granted this does affect the biome outside of expeditions as well but you can at least admeme using it without worrying about people getting round removed due to lag and stuff.

**Comment by Necro-a** [comment]
> Lots of the same stuff. Have not tested this in-game yet. (How-to-test steps would be appreciated.)

I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.

**Comment by TsjipTsjip** [comment]
> I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.

I am not a salvager. I have not played salvage since months ago. Tell me how to test this biome in-game, what do I need to do in-game to test it.

**Comment by Necro-a** [comment]
> > I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.
> 
> I am not a salvager. I have not played salvage since months ago. Tell me how to test this biome in-game, what do I need to do in-game to test it.

As in, doing an actual ingame expedition?

**Comment by Necro-a** [comment]
> > I believe it should be as easy as putting the geneticore atlas in maps/dungeon and the other geneticore.yml file in procedural/themes, then running the dungen command.
> 
> I am not a salvager. I have not played salvage since months ago. Tell me how to test this biome in-game, what do I need to do in-game to test it.

1) Download the following: 
/Maps/_Umbra/Dungeon/geneticore.yml 
/Prototypes/_Umbra/Procedural/Themes/geneticore.yml
/Prototypes/_Umbra/Procedural/dungeon_configs.yml
/Prototypes/_Umbra/Procedural/salvage_mods.yml
/Prototypes/_Umbra/tags.yml

2) Put the files in the appropriate areas. (Aka just copy the file paths for the folders

3) Edit biome_templates.yml the same way its changed in the commits

4) Load the game up, spawn an expedition console, pick a shadow biome expedition.

5) Give yourself whatever gear you deem appropriate for a salvager to have, spawn a reclaimer (or any shuttle that can FTL.) and FTL to the expedition and play through it

Hope that helps.

**Comment by TsjipTsjip** [comment]
Okay I didn't mean *that* in depth. What I wanted to know is if there's a way to force an expedition to generate with these settings, rather than having to get lucky on the expeditions console. (Far as I understand it, even shadow biome could be the regular shadow biome and thus not your facility.) I'll see if I can review this throughout next week. Timeline for merge is probably gonna be next sunday if no more major issues are found.

File: Resources/Prototypes/Procedural/salvage_mods.yml [hunk 1]
```diff
@@ -248,3 +248,5 @@
   proto: Haunted
   biomes:
     - Caves
+
+
```
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/salvage_mods.yml:252]: This file should be left unchanged.

File: Resources/Prototypes/Procedural/biome_templates.yml [hunk 2]
```diff
@@ -504,7 +504,7 @@
     - !type:BiomeEntityLayer
       allowedTiles:
       - FloorChromite
-      threshold: 0.2
+      threshold: 0.5
```
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/biome_templates.yml:507]: Add a `# Umbra` comment.

File: Resources/Prototypes/Procedural/biome_templates.yml [hunk 3]
```diff
@@ -513,7 +513,7 @@
         lacunarity: 2
         gain: 0.4
       entities:
-      - FloorChromiteChasm
+      - ShadowKudzuWeak
```
Comment (TsjipTsjip) [inline Resources/Prototypes/Procedural/biome_templates.yml:516]: Add a `# Umbra` comment.

**Review by TsjipTsjip — CHANGES_REQUESTED** [review]
More minor remarks, this is already ready for full review tho. I'm gonna see if I can get that done throughout the following week.

**Comment by Necro-a** [comment]
> Okay I didn't mean _that_ in depth. What I wanted to know is if there's a way to force an expedition to generate with these settings, rather than having to get lucky on the expeditions console. (Far as I understand it, even shadow biome could be the regular shadow biome and thus not your facility.) I'll see if I can review this throughout next week. Timeline for merge is probably gonna be next sunday if no more major issues are found.

You can forcefully generate a _dungeon_ but an entire expedition im not sure. Also, no worries, it should always generate a facility no matter what. If you wanna make sure you don't have to be lucky with the expedition console you can try removing the biome mods and dungeon mods from salvage_mods.yml so it ONLY shows the shadow biome.

**Comment by SlamBamActionman** [comment]
Tested ingame and it seems to run just fine. 👍👍

There's probably something to be said about how the Elemental category is a pretty weak enemy type (especially since Salvage is the one job fully equipped to deal with ore crabs), but other than that it seems coolio.

**Comment by Necro-a** [comment]
> Tested ingame and it seems to run just fine. 👍👍
> 
> There's probably something to be said about how the Elemental category is a pretty weak enemy type (especially since Salvage is the one job fully equipped to deal with ore crabs), but other than that it seems coolio.

Understood. Should i maybe add a salvage version of the ore crabs that are a bit stronger? Or maybe add specific types of reagent slimes to it so it matches in difficulty with the other enemy types?

**Comment by TsjipTsjip** [comment]
Will have a proper look at this soon.

**Review by luckyshotpictures — APPROVED** [review]
Requested changes have been provided. Good to merge.

**Review by TsjipTsjip — APPROVED** [review]
Works in-game, drops were fixed. Sure, let's do this.

**Pull request merged**